### PR TITLE
Add code coverage analysis

### PR DESCRIPTION
**Pull request by ivan-podorozhnyi**
Added requirements-dev.txt for Python dependencies with code coverage.

Updated `verify_project.sh` script to run tests with code coverage.

Updated `README.md`.

Updated `.travis.yml`.

Affects #3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,11 +7,12 @@ A demo of Python's project with automated tests.
 How to install?
 ===============
 1. Create separate virtual environment with Python 3.5.x.
-2. Run `pip install -r requirements.txt` to install required Python's dependencies.
+2. Run `pip install -r requirements-dev.txt` to install required Python's dependencies.
```
Comment (extsoft) [inline README.md]: We have to support two installations:
- for run - use `requirements.txt`
- for development - use `requirements.txt` and `requirements-dev.txt`

File: README.md [hunk 2]
```diff
@@ -7,11 +7,12 @@ A demo of Python's project with automated tests.
...
 
-Howe to run automated tests?
+How to run automated tests?
 ============================
-Run `py.test demo_python_at` within your virtual environment.
+1. Run `py.test demo_python_at` within your virtual environment.
+2. Run `py.test --cov=demo_python_at tests/` within your virtual environment to run tests with coverage.
```
Comment (extsoft) [inline README.md]: This command runs unit tests, but this section about automated tests in `demo_python_at`. Please restore this paragraph.

File: requirements-dev.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+coverage==4.4.1
+py==1.4.34
+pytest==3.1.1
```
Comment (extsoft) [inline requirements-dev.txt]: This dependency is specified in `requirements.txt`. Don't duplicate dependencies.

File: requirements-dev.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+coverage==4.4.1
```
Comment (extsoft) [inline requirements-dev.txt]: Don't add all output from `pip freeze`. Please add only packages that you installed with `pip install`.

**Review by extsoft — CHANGES_REQUESTED** [review]
@ivan-podorozhnyi please fix my comments.

**Comment by ivan-podorozhnyi** [comment]
@extsoft Please check the changes.

File: verify_project.sh [hunk 5]
```diff
@@ -1,6 +1,10 @@
 #!/usr/bin/env bash
 set -e
 
-py.test tests
+if command command coverage >/dev/null 2>&1 ; then
```
Comment (AntonKarabaza) [inline verify_project.sh]: @ivan-podorozhnyi Please, fix `command command coverage` call.

File: .travis.yml [hunk 6]
```diff
@@ -2,6 +2,7 @@ language: python
 python: "3.5"
 install:
   - pip install -r requirements.txt
+  - pip install -r requirements-dev.txt
```
Comment (AntonKarabaza) [inline .travis.yml]: Please, in commit message specify that dependencies will be installed both from `requirements.txt` and  `requirements-dev.txt`.

File: verify_project.sh (hunk 5 above)
Comment (AntonKarabaza) [inline verify_project.sh]: Please, fix `command command coverage` call.

File: .travis.yml (hunk 6 above)
Comment (AntonKarabaza) [inline .travis.yml]: Please, in commit message specify that dependencies will be installed both from `requirements.txt` and  `requirements-dev.txt`.

**Review by AntonKarabaza — CHANGES_REQUESTED** [review]
@ivan-podorozhnyi Take a look at my comments.

**Comment by ivan-podorozhnyi** [comment]
@AntonKarabaza Please check my last changes.

**Review by AntonKarabaza — APPROVED** [review]

**Comment by AntonKarabaza** [comment]
@ivan-podorozhnyi  Good. Thank you!

File: requirements-dev.txt [hunk 7]
```diff
@@ -0,0 +1,2 @@
...
```
Comment (extsoft) [inline requirements-dev.txt]: Do we really need `coverage` here?

File: README.md [hunk 8]
```diff
@@ -7,9 +7,11 @@ A demo of Python's project with automated tests.
 How to install?
 ===============
 1. Create separate virtual environment with Python 3.5.x.
-2. Run `pip install -r requirements.txt` to install required Python's dependencies.
+2. Run `pip install -r requirements.txt` or `pip install -r requirements-dev.txt`  to install required Python's dependencies.
```
Comment (extsoft) [inline README.md]: `How to install?` section is about automated tests. Why do you add development details here?

File: verify_project.sh [hunk 9]
```diff
@@ -1,6 +1,10 @@
 #!/usr/bin/env bash
 set -e
 
-py.test tests
+if command coverage >/dev/null 2>&1 ; then
```
Comment (extsoft) [inline verify_project.sh]: We have to measure coverage each time we run unit tests.

**Review by extsoft — CHANGES_REQUESTED** [review]
@ivan-podorozhnyi 
Some additional comments:
- all commits except 76114e1c8e3c0261219461810fa7242c98445698 have too general subject

File: requirements-dev.txt (hunk 7 above)
Comment (ivan-podorozhnyi) [reply requirements-dev.txt]: @extsoft Moved coverage and pytest-cov dependencies to `requirements.txt`.

File: README.md (hunk 8 above)
Comment (ivan-podorozhnyi) [reply README.md]: @extsoft Removed everything.

File: verify_project.sh (hunk 9 above)
Comment (ivan-podorozhnyi) [reply verify_project.sh]: @extsoft Clear, now all tests run with coverage.

File: verify_project.sh [hunk 10]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 set -e
 
-py.test tests
 flake8 demo_python_at/
+py.test --cov=demo_python_at
```
Comment (extsoft) [inline verify_project.sh]: Unit testing has to be first.

File: requirements.txt [hunk 11]
```diff
@@ -1 +1,3 @@
 pytest==3.1.1
+coverage==4.4.1
```
Comment (extsoft) [inline requirements.txt]: pip installs `coverage` automatically when it installs `pytest-cov`. Please remove `coverage` from the requirements.txt.

File: requirements.txt [hunk 12]
```diff
@@ -1 +1,3 @@
...
+pytest-cov==2.5.1
```
Comment (extsoft) [inline requirements.txt]: This has to migrate to `requirements-dev.txt` as this is only dev dependency.

**Review by extsoft — CHANGES_REQUESTED** [review]
@ivan-podorozhnyi please work on my comments.

File: requirements.txt (hunk 12 above)
Comment (ivan-podorozhnyi) [reply requirements.txt]: @extsoft But you said that we have to measure coverage each time we run unit tests.
command `py.test --cov=demo_python_at` runs all tests inside our projects with coverage.
Do I have to return back to version with installed library check?
```
if command coverage >/dev/null 2>&1 ; then
    py.test --cov=demo_python_at
else 
    py.test tests
fi
```

Comment (extsoft) [reply requirements.txt]: We have to run unit test only: `py.test tests`. `verify_project.sh` is used for development verifying, not to run automated tests.

Comment (ivan-podorozhnyi) [reply requirements.txt]: @extsoft Understood, thanks.

**Comment by ivan-podorozhnyi** [comment]
@extsoft Now tests run before flake8 checks, pytest-cov moved to dev dependencies.

**Review by extsoft — APPROVED** [review]

**Pull request merged**